### ISSUE TITLE
'Save solution on build' when triggered from the command line

### ISSUE THREAD

**Issue by NoelAbrahams** [madskristensen/Tweakster#40]
Is there any way to get this to run from the command line?

eg myCommandFile.cmd
```cmd
@echo off

REM Locate the path to `msbuild`
for /f "usebackq tokens=*" %%i in (`"%ProgramFiles(x86)%\Microsoft Visual Studio\Installer\vswhere.exe" -latest -requires Microsoft.Component.MSBuild -find MSBuild\**\Bin\MSBuild.exe`) do (
 
  echo Save solution before build
   >>>> HERE >>>
   
   echo Building the solution...
   "%%i" -m -verbosity:quiet -nologo

   echo Complete
  exit /b !errorlevel!
)
```

I understand the difficulty since Visual Studio is a separate process to the command line.

The command above is actually bound to an 'External Tool' (tools menu)  and then triggered by the user clicking on the menu item in VS.

The problem is when there are open files with unsaved changes, the command runs on the file persisted on disk ignoring the changes.

Is there any way to get Visual Studio to save the solution before build for this scenario? Perhaps an option on the 'External Tools' menu?

![image](https://user-images.githubusercontent.com/1106823/93666352-7dfbe800-fa75-11ea-9797-fe39b61a2044.png)

**Comment by madskristensen** [comment]
This is an MSBuild command and not one done in Visual Studio. I don't know what you are trying to achieve here. Do you wish to replace the regular build in VS with one you run on the command line instead?

**Comment by NoelAbrahams** [comment]
> This is an MSBuild command and not one done in Visual Studio. 

Yes, it is clearly an MSBuild script. The script is saved as a file. This file is triggered as an 'External Tool' as stated here:

> The command above is actually bound to an 'External Tool' (tools menu) and then triggered by the user clicking on the menu item in VS.

Regarding this:

>  I don't know what you are trying to achieve here

The ultimate goal is the following:
* Given a solution with say 50 projects.
* Build a subset of projects.
* The projects may need to be built in a certain order (since they there are hierarchical dependencies).

I don't want to be running things from the command line. This build should be triggered from within Visual Studio. Creating a script and adding it as an 'External Tool' achieves this.

The only thing that is failing is that when there are files with unsaved changes, they are not saved when the build is run.

Is the problem clear enough?

**Comment by madskristensen** [comment]
yes, thanks

**Comment by madskristensen** [comment]
All feature requests to this extension must be backed by a real Visual Studio suggestion registered at [Developer Community](https://developercommunity.visualstudio.com/content/idea/post.html?space=8). Please create one and post the link back here.
